### DISCUSSION TITLE
Config to disable "stackable" upgrades (specifically the stack size multiplier)

### DISCUSSION THREAD

**Discussion by mochalit** [P3pp3rF1y/SophisticatedBackpacks#1074] [Ideas]
i help run a fairly large server which has this mod. to avoid possible integer overload issues, we have disabled all tiers of stack upgrades apart from the first tier. recently we have found out that you can achieve the same effect by placing multiple of the tier one stack upgrade and let it multiply exponentially. 
it would be very beneficial for us if there was a config to disable this so that the tier one stack upgrade only applies its effect once. if not, we may have to disable the stacking upgrades entirely to avoid crashes which would be very unfortunate. please let me know if we are missing something that would already allow us this. thank you
![image](https://github.com/P3pp3rF1y/SophisticatedBackpacks/assets/151514193/0e0c205a-b337-4610-aabe-3c6784e9de9f)

**Comment by P3pp3rF1y** [comment]
What kind of integer overflow and crashes are you referring to? The mod is coded to not allow more than integer max to be put in a slot and I am also not aware of any crash related to stack upgrades?

**Reply by mochalit** [reply]
i apologize, i was mistaken.
 i've been informed the issue is the size of the player data files and how having all the possibly thousands of  items in the backpack makes the size bigger and eats up storage space. i would still like to request the feature be added for this.

**Reply by P3pp3rF1y** [reply]
and that is false as well, SB doesn't save backpack contents in playerdata so that data overflowing can't happen and even if it was saved there the stack size effect on overflow would be so miniscule that it is completely irrelevant. What matters way more there is number of slots in inventory a bit and primarily size of individual nbt. 
Stack Size - The way SB saves size of stacks is in int type instead of short which means that it takes 2 bytes more per slot which if you had 200 slots would go up to 800 bytes, but you can't even have that many by default so it's probably something like 500 bytes at max.
Number of slots - each item needs to store multiple bits of information. Extremely simple item with no nbt will probably take up something like 40 bytes (20 bytes for its registry name + 4 bytes for size + some change for names of the attributes) so by adding 20 slots you are already at the extra size required for storing max int stack size in backpacks.
But Nbt blows these two out of the water. there are items that save a lot of nbt on them like spawners, bees, all kinds of entity catch / respawn things, ... with these a single item can easily have more than 1,000 bytes of nbt on it and if you start putting multiple of these items in single backpack it just gets worse.

That said player data isn't used for SB contents, they are stored separately and in the near future with neoforge even the limit that is there for player data and other is going to be handled so this won't be concern anymore at all.

**Comment by Zunwolf1896** [comment]
Hello, sorry for the very delayed response, been very busy recently. 

I am the owner of the server Mocha is talkign about and it seems the information I recieved was wrong - the person must have mixed up the NBT stuff with stack size. Thank you very much for clearing this up. 

Would it still be possible to add an option in the configs to make it where you can only have one type of an upgrade in your backpacks active slots at a time? - for example if you have a stackable upgrade, you can not add another stackable upgrade, or if you have a XP storage, you can not have a second one?
